src/environmentd/src/deployment/preflight.rs

```diff
@@ -227,6 +229,8 @@ pub async fn preflight_0dt(
 /// Check if there have been any DDL that create new collections or replicas,
 /// restart in read-only mode if so, in order to pick up those new items and
 /// start hydrating them before cutting over.
+///
+/// We do this by enumerating the set of IDs that need to be allocated after the ddl check began and seeing if any were committed to the catalog.
```

Comment:
I don't quite follow the comment, specifically: "the set of IDs that need to be allocated after the ddl check began". I think it should be "the set of IDs that were allocated after the preflight check begin"?